src/support_sphere_py/tests/resources/scripts/update_db_sample_data.py

```diff
@@ -68,6 +68,41 @@ def populate_cluster_and_household_details():
     household = Household(cluster=all_clusters[-1], name="Household1")
     BaseRepository.add(household)
 
+def populate_real_cluster_and_household():
+    """
+    Populate clusters and households based on household data container cluster name and address.
+    During the creation of household, random signup code is also generated using uuid.
+    """
+    household_data = Path("./support_sphere_py/tests/resources/data/households.csv")
+    with household_data.open(mode='r', newline='') as file:
+        csv_reader = csv.DictReader(file)
+
+        cluster_uids = {}
+        for row in csv_reader:
+            # Get and set cluster
+            cluster_name = row["CLUSTER"]
+            if cluster_name not in cluster_uids:
+                cluster = Cluster(name=cluster_name)
+                cluster_id = cluster.id
+                cluster_uids[cluster_name] = cluster.id
+                
+                # Add cluster to the database
+                BaseRepository.add(cluster)
+            else:
+                cluster_id = cluster_uids[cluster_name]
+
+            # Setup household
+            household_address = row['ADDRESS']
+            household = Household(cluster_id=cluster_id, address=household_address)
+            
+            # Generate random signup code
+            uid = uuid.uuid4()
+            code = uid.hex[:7].upper()
+            signup_code = SignupCode(code=code, household_id=household.id)
```

Comment:
It's extremely unlikely, but given that there are fewer bits of entropy with a truncated UUID, should we make sure we're able to see and handle collisions here?

Also I'm wondering if, for testing, having the signup codes be random every time the db starts up might make it harder to verify things (unless we have some specific dummy single-/multi-family households whose codes are the same every time (in dev dbs only). Not a major concern or anything! Just wondering if this is worth considering is all

Reply:
> It's extremely unlikely, but given that there are fewer bits of entropy with a truncated UUID, should we make sure we're able to see and handle collisions here?

Since the field is a primary key, the database will complain if the value is not unique during insert, so we have that check.

> (unless we have some specific dummy single-/multi-family households whose codes are the same every time (in dev dbs only)

Yea, it's easy enough for us to create dummy codes :)

Reply:
> Since the field is a primary key, the database will complain if the value is not unique during insert, so we have that check.

That's good to have that check, but what happens when/if that check fails? Does the sample data population script halt and exit partially completed? Does it throw an error that we can catch and retry on? Should we try/catch that possibility here?

Reply:
Yea it'll throw an error and exit.. You brought up a good point to maybe it should retry when this happens. Okay, I'll update the code to account for this 😛

Reply:
Alright, sounds good! In the very rare case that this does happen in the future, I think that that retry could save quite a lot of debugging time :)